src/snowplow/emitter.lua

```diff
@@ -0,0 +1,157 @@
+    return false, "Unrecognised status code [" .. ss(status_code) .. "]"
+  elseif code >= 400 and code < 500 then
+    return false, "HTTP status code [" .. ss(status_code) .. "] is a client error"
+  elseif code >= 500 then
+    return false, "HTTP status code [" .. ss(status_code) .. "] is a server error"
+  end
+
+  return true
+end
+
+-- Performs a GET request with a given payload to the collector.
+-- @param uri string: The URI (including querystring) to GET
+-- @return boolean, string: Whether event was successfully collected; and the reason for failure if not
+local function _get(self, payload)
+  local url = self.collector_url .. payload
+  -- `resp` is the table `:getinfo` reads from
+  local resp = {}
+  local c = curl.easy({
+    url = url,
+  }):setopt_writefunction(table.insert, resp)
+
+  local _, err = pcall(function()
+    c:perform()
+  end)
+
+  if err ~= nil then
+    return false, "Host [" .. self.collector_url .. payload .. "] not found (possible connectivity error)"
+  end
+
+  return handle_status_code(c:getinfo(curl.INFO_RESPONSE_CODE))
+end
+
+-- Performs a POST request with a given payload to the collector.
+-- @param uri string: The URI (including querystring) to GET
+-- @param method string: The HTTP method to use
+-- @return boolean, string: Whether event was successfully collected; and the reason for failure if not
+local function _post(self, payload)
+  -- `resp` is the table `:getinfo` reads from
+  local resp = {}
+  local c = curl.easy({
+    url = self.collector_url,
+    post = true,
+    httpheader = {
+      "Content-Type: application/json",
+    },
+    postfields = payload,
+  }):setopt_writefunction(table.insert, resp)
+
+  local _, err = pcall(function()
+    c:perform()
+  end)
+
+  if err ~= nil then
+    return false, "Host [" .. self.collector_url .. "] not found (possible connectivity error)"
+  end
+
+  return handle_status_code(c:getinfo(curl.INFO_RESPONSE_CODE))
+end
+
+-- --------------------------------------------------------------
+-- Public methods
+
+-- Returns the request method
+-- @return string: The request method
+function Emitter:get_request_method()
+  return self.request_method
+end
+
+-- Returns the collector url
+-- @return string: The collector url
+function Emitter:get_collector_url()
+  return self.collector_url
+end
+
+-- Sends a payload to the collector
+-- @param payload string: The payload to send
+-- @return boolean, string: Whether event was successfully collected; and the reason for failure if not
+function Emitter:send(payload)
+  return self.request_func(self, payload)
+end
+
+-- Builds a new emitter instance
+-- @param collector_url string: The url of the collector
+-- @param request_method string: The request method to use
+-- @return Emitter: The new emitter instance
+function emitter.new(collector_url, request_method)
+  local e = {}
+  setmetatable(e, { __index = Emitter })
+  e.collector_url = collector_url
```

Comment:
I assume this is not neccessary as the actual `collector_url` is set on line 153?